### ISSUE TITLE
Do not overwrite the selected block when pasting another block

### ISSUE THREAD

**Issue by luisherranz** [WordPress/gutenberg#68102]
### Description

In my opinion, when copy-pasting blocks, it is dangerous/confusing that the selected block is overwritten.

Ideally, this should only happen when the selected block is an empty paragraph. If not, the block being pasted should be inserted right after the selected block.

### Step-by-step reproduction instructions

1. Copy a block using CTRL-C or CMD-C. 
2. Select another block. 
3. Paste the initial block using CTRL-V or CMD-V. The selected block disappears.

### Screenshots, screen recording, code snippet

https://github.com/user-attachments/assets/4a39442e-9720-47b9-bc07-be8e853aa591

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by Mayank-Tripathi32** [comment]
@luisherranz I agree that from a user’s perspective, it can be frustrating not to be able to paste components next to existing ones for duplication. Perhaps we could introduce a popup when a user attempts this action, offering options such as replacing the current block or adding the new component next to it. This could enhance the overall user experience by making duplication more intuitive.

**Comment by ellatrix** [comment]
Sure, we can try that. It's funny that you bring this up, I was just exploring something for pasting images onto the cover and media and text blocks that preserves the block and adds the image to the block instead.

**Comment by luisherranz** [comment]
Honestly, I believe introducing a popup might be a bit annoying for the user and disrupt their flow when they are copying and pasting many blocks due to some block re-organization. But that's my opinion 🙂

**Comment by yogeshbhutkar** [comment]
Displaying a popup every time content is pasted might not be appreciated by a lot of users. Instead, we can take inspiration from the behavior of the `Heading` and `Paragraph` blocks for a more seamless user experience.

https://github.com/user-attachments/assets/949a55d4-8acc-4e8d-9709-eac8a512dab5

When copying and pasting a paragraph onto a heading block, the content of the paragraph is applied to the heading, and the paragraph block becomes a new block inserted below it. The previous heading is then stored within this new block. The Heading block is only replaced if it is explicitly selected by double-clicking.

We could implement a similar approach (without carry forwarding the content, of course). Here's how it might work:  

- A **single click** while pasting onto a block would shallow-select it, creating a new block in its place and moving the previous block below.  
- A **double click** would confirm the block selection and replace its content upon pasting.  

While some users might find this behavior annoying, it's an established design pattern already in use with Heading and Paragraph blocks. To accommodate preferences, we could include a toggle in the settings to enable or disable this advanced copy-paste feature. Over time, this approach could feel intuitive and enhance usability with regular use.

But then the double click + paste is used by containers to place the pasted elements within it. We might need to prioritize either of the behaviors I guess.
